docs/book/src/images/cluster-topology-reconciller.puml

```diff
@@ -0,0 +1,50 @@
+@startuml
+'https://plantuml.com/component-diagram
+
+
+package "ClusterClass" as cc {
+  [Patches]
+  [Variable Schemas]
+  [Template 1]
+  [Template 2]
+  [\t...]
+
+  [Patches] -[hidden]-> [Variable Schemas]
+  [Variable Schemas] -[hidden]-> [Template 1]
+  [Template 1] -[hidden]-> [Template 2]
+  [Template 2] -[hidden]-> [\t...]
+}
+
+package "Cluster" as cluster {
+  [Variables]
+  [InfraCluster]
+  [ControlPlane]
+  [ ...]
+}
+
+
+package "ClusterTopologyReconciler" as ctr {
+  [1. getBluePrint] as bp
+  [2. getCurrentState] as cs
+  [3. computeDesiredState] as ds
+  [4. reconcileState] as rs
+
+  bp -[hidden]-> cs
+  cs -[hidden]-> ds
+  ds -[hidden]> rs
+
+    package "DesiredCluster" as dc {
+      [Desired Variables]
```

Comment:
If I understood correctly those are the results of computeDesiredState. If yes, let's name that `Desired Cluster` as:
* the cluster itself is reconciled
* the variables are not

Reply:
makes sense. I didn't get the semantic behind variable and cluster.  thanks for pointing it out. 

A question, what part of the cluster spec is considered as "variables"?  Or am I misunderstanding your intention in the original diagram?

Reply:
@klxfeiyang Variables are Cluster.spec.topology.variables